tests/interactive/test_app.py

```diff
@@ -147,54 +147,54 @@ async def test_buttons():
-          %two = arith.constant 2 : index
-          %res = arith.muli %n, %two : index
-          func.return %res : index
-        }
-        """
+        # Select two passes
+        app.pass_pipeline = tuple(
+            (
+                *app.pass_pipeline,
+                convert_func_to_riscv_func.ConvertFuncToRiscvFuncPass,
+                convert_arith_to_riscv.ConvertArithToRiscvPass,
+            )
         )
+
+        # assert that pass selection affected Output Text Area
+        await pilot.pause()
         assert (
             app.output_text_area.text
             == """builtin.module {
-  func.func @hello(%n : index) -> index {
-    %two = arith.constant 2 : index
-    %res = arith.muli %n, %two : index
-    func.return %res : index
+  riscv.assembly_section ".text" {
+    riscv.directive ".globl" "hello"
+    riscv.directive ".p2align" "2"
+    riscv_func.func @hello(%n : !riscv.reg<a0>) -> !riscv.reg<a0> {
+      %0 = riscv.mv %n : (!riscv.reg<a0>) -> !riscv.reg<>
+      %n_1 = builtin.unrealized_conversion_cast %0 : !riscv.reg<> to index
+      %two = riscv.li 2 : () -> !riscv.reg<>
+      %two_1 = builtin.unrealized_conversion_cast %two : !riscv.reg<> to index
+      %res = builtin.unrealized_conversion_cast %n_1 : index to !riscv.reg<>
+      %res_1 = builtin.unrealized_conversion_cast %two_1 : index to !riscv.reg<>
+      %res_2 = riscv.mul %res, %res_1 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+      %res_3 = builtin.unrealized_conversion_cast %res_2 : !riscv.reg<> to index
+      %1 = builtin.unrealized_conversion_cast %res_3 : index to !riscv.reg<>
+      %2 = riscv.mv %1 : (!riscv.reg<>) -> !riscv.reg<a0>
+      riscv_func.return %2 : !riscv.reg<a0>
+    }
   }
 }
 """
         )
 
-        # Test clicking the "clear input" button
-        await pilot.click("#clear_input_button")
-
-        # assert that the input text area has been cleared
+        # press "Remove Last Pass" button
+        await pilot.click("#remove_last_pass_button")
         await pilot.pause()
-        assert app.input_text_area.text == ""
 
-        app.input_text_area.insert(
-            """
-        func.func @hello(%n : index) -> index {
-          %two = arith.constant 2 : index
-          %res = arith.muli %n, %two : index
-          func.return %res : index
-        }
-        """
-        )
-
-        # Select a pass
-        app.pass_pipeline = tuple(
-            (*app.pass_pipeline, convert_func_to_riscv_func.ConvertFuncToRiscvFuncPass)
+        # assert that the pass pipeline and Output Text Area change accordingly
+        expected_pass_pipeline = ()
+        expected_pass_pipeline = tuple(
+            (
+                *expected_pass_pipeline,
+                convert_func_to_riscv_func.ConvertFuncToRiscvFuncPass,
+            )
         )
-
-        # assert that pass selection affected Output Text Area
-        await pilot.pause()
+        assert app.pass_pipeline == expected_pass_pipeline
```

Comment:
nit:
```suggestion
        # press "Remove Last Pass" button
        current_pipeline = app.pass_pipeline
        await pilot.click("#remove_last_pass_button")
        await pilot.pause()
        assert app.pass_pipeline == current_pipeline[:-1]
```
feels a tiny bit neater to me